core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,159 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.common;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.CreateStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods to help create and manage Kinesis Streams
+ */
+public final class KinesisUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisUtil.class);
+  private static final long DELETE_TIMEOUT = 1000 * 120;
+  private static final long CREATE_TIMEOUT = 3 * 60 * 1000;
+  private static final long SLEEP_INTERVAL = 1000 * 10;
+
+  private static final String STATUS_CREATING = "CREATING";
+  private static final String STATUS_DELETING = "DELETING";
+  private static final String STATUS_ACTIVE = "ACTIVE";
+  private static final String STATUS_UPDATING = "UPDATING";
+
+
+  private KinesisUtil() {
+  }
+
+  /**
+   * Creates an Amazon Kinesis stream if it does not exist and waits for it to become available
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read and write privileges
+   * @param streamName The Amazon Kinesis stream name to create
+   * @param shardCount The shard count to create the stream with
+   * @throws IllegalStateException Invalid Amazon Kinesis stream state
+   * @throws IllegalStateException Stream does not become active before the timeout
+   */
+  public static void createAndWaitForStream(AmazonKinesisClient kinesisClient,
+                                            String streamName,
+                                            int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      String state = getStreamState(kinesisClient, streamName);
+      switch (state) {
+        case STATUS_DELETING:
+          long waitTimeDelete = System.currentTimeMillis() + DELETE_TIMEOUT;
+          while (System.currentTimeMillis() < waitTimeDelete && streamExists(kinesisClient, streamName)) {
+            try {
+              LOG.debug("...Deleting Stream {} ...", streamName);
+              Thread.sleep(SLEEP_INTERVAL);
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+            }
+          }
+          if (streamExists(kinesisClient, streamName)) {
+            throw new IllegalStateException(String.format("KinesisUtils timed out waiting for stream {} to delete",
+                                                          streamName));
+          }
+          createStream(streamName, shardCount, kinesisClient);
+          break;
+        case STATUS_ACTIVE:
+          LOG.info("Stream {} is ACTIVE", streamName);
```

Comment:
since ACTIVE/UPDATING are Kinesis-internal statuses,  better to say ``Stream {} already exists`` here and ``Stream {} is being updated`` on line 84?